src/preprocess.js

```diff
@@ -56,6 +56,45 @@ ${diagramText}
   return JSON.stringify(diagramObject)
 }
 
+// @ts-check
+/**
+ * @param {string} diagramText
+ * @param {any} context
+ * @returns {string}
+ */
+module.exports.preprocessPlantUML = function (diagramText, context) {
+  // see: http://plantuml.com/en/preprocessing
+  const RegExInclude = /^(\s*)!(include(?:sub)?)\s+(.+?)(?:!(\w+))?$/i

```

Comment:
You can take the following (more complete) Java regular expression from Kroki: https://github.com/yuzutech/kroki/blob/3309c057f1dfbfcb6ea58c1a717119f93db4a4ad/server/src/main/java/io/kroki/server/service/Plantuml.java#L72

A few notes:
- We should ignore includes from the stdlib (since the stdlib is included in the PlantUML jar and available on the Kroki server) or at least we should not thrown an error: `!include <some/stdlib>`
- You can have a trailing comment `# this is a comment`.
- If we want to resolve include properly we will need to implement the same logic regarding the "subid" feature (ie. `!include file.puml!2` will include the 3rd document in a file `file.puml`. Please note that it's also possible to use a custom id:

> For example, if you use !include foo.txt!1, the second @startuml/@enduml block within foo.txt will be included.
You can also put an id to some @startuml/@enduml text block in an included file using @startuml(id=MY_OWN_ID) syntax and then include the block adding !MY_OWN_ID when including the file, so using something like !include foo.txt!MY_OWN_ID.

Basically we will need an ultra-simplified PlantUML parser to read the file and extract the `@startuml/@enduml` blocks. I think we can leave this feature for now and issue a warning.

Reply:
Thanks!

I've changed this PR to WIP/draft and will add all the needed include features, because i've  faced problems with our PlantUML files in some projects (std-lib, recursive includes etc.) and cannot use this basic not complete version ;)

- use the more complete regex
- skip lines with `!include <std/lib/path>`
- add `!include file!index` and `!include file!id`
- add recursive include support and prevent infinite include loop
- i'm not sure about add `!includesub id` and `!includeurl url`, because we are not using it and it's deprecated, but may be for compatibility we should add this...

Reply:
> i'm not sure about add !includesub id and !includeurl url, because we are not using it and it's deprecated, but may be for compatibility we should add this...

yes until PlantUML removes them completely we will need to support `includesub` and `includeurl`.

Reply:
I've adressed the issues, please review